### ISSUE TITLE
[Google3 Bug]: [a11y] [Android] Talkback can't focus text sections with inline links on Android mobile web

### ISSUE THREAD

**Issue by math1man** [flutter/flutter#174712]
### Help us understand the severity of this issue

- [x] causing severe production issues e.g. malfunctions or data loss
- [ ] blocking next binary release
- [ ] blocking a client feature launch within a quarter
- [ ] nice-to-have but does not block a launch within the next quarter

### Steps to reproduce

1. Create a flutter web app that creates a focusable inline link, using `RichText`. Our app uses https://pub.dev/packages/tagged_text for this, but the minimal repro below uses an even simpler implementation derived from it.
2. Open the web app on an Android device in Chrome
3. Enable Android TalkBack
4. Enable Flutter accessibility
5. Observe how Talkback interacts with the app

Steps 1-2 can also be achieved by opening https://dartpad.dev/?id=0239957b1f8428166a22f1565932b6f0&channel=main in Chrome on an Android device and running the dartpad.

### Expected results

The non-link portion of the `RichText` should be focusable using talkback.

### Actual results

Only the link is focusable.

### Code sample

<details open><summary>Code sample</summary>

Also available at https://gist.github.com/math1man/0239957b1f8428166a22f1565932b6f0 and https://dartpad.dev/?id=0239957b1f8428166a22f1565932b6f0&channel=main

```dart
import 'package:flutter/material.dart';

void main() {
  runApp(const MyApp());
}

class MyApp extends StatelessWidget {
  const MyApp({super.key});

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      debugShowCheckedModeBanner: false,
      home: Scaffold(
        body: ListView(
          children: [
            Text('Start'),
            Builder(
              builder: (context) => RichText(
                text: TextSpan(
                  children: [
                    TextSpan(text: 'Hello, World! '),
                    WidgetSpan(
                      alignment: PlaceholderAlignment.middle,
                      child: _Link(
                        text: 'Link',
                        onTap: () {
                          ScaffoldMessenger.of(
                            context,
                          ).showSnackBar(SnackBar(content: Text('Clicked')));
                        },
                      ),
                    ),
                  ],
                ),
              ),
            ),
            Text('End'),
          ],
        ),
      ),
    );
  }
}

class _Link extends StatelessWidget {
  const _Link({required this.text, this.onTap});

  final String text;
  final GestureTapCallback? onTap;

  @override
  Widget build(BuildContext context) {
    final colors = Theme.of(context).colorScheme;
    return Semantics(
      link: true,
      linkUrl: Uri.parse('#'),
      container: true,
      onTap: onTap,
      child: GestureDetector(
        onTap: onTap,
        child: Text(
          text,
          style: TextStyle(
            color: colors.primary,
            decoration: TextDecoration.underline,
            decorationColor: colors.primary,
          ),
        ),
      ),
    );
  }
}
```

</details>

### Screenshots or Video

<details open>
<summary>Screenshots / Video demonstration</summary>

https://github.com/user-attachments/assets/a05b4a07-851e-4740-bf85-dea3625e8643

</details>

### Logs

Logs: N/A

### Flutter Doctor output

Doctor output: N/A

**Comment by math1man** [comment]
Internal bug b/440396024. Client: `huggsy`

**Comment by math1man** [comment]
Some added context: we do have a regular Android app where things work normally, but we also have a high-usage surface where our web app is accessed via an Android WebView, and that use case suffers from this bug same as running in Chrome.

**Comment by tirth-patel-nc** [comment]
@math1man thanks for the report. I’m seeing the same behavior with the latest SDK versions on web using mobile chrome (pixel 6a). However it doesn't replicate on Deskop-Web (Chrome) with Voice-Over in Mac.

```
stable : 3.35.2
master : 3.36.0-1.0.pre-366
```

#129033 might be related.

**Comment by math1man** [comment]
Yep, that's consistent with my experience. The issue only occurs on Android-Web (both Chrome app and system WebView, which is also powered by Chrome).

**Comment by chunhtai** [comment]
This is due to the label gets merged up. Will likely need this https://github.com/flutter/flutter/issues/166095 in order to properly fix this issue
